application/features/authentication.feature

```diff
@@ -177,6 +177,37 @@ Feature: Authentication
     When I request "/authentication" be created
     Then the response status code should be 200
 
-#    TODO: attempt to authenticate a user who doesn't have a password yet, expect 400 (ensure timing attack protection is enforced)
+  Scenario: Check "nag" flags on user resource in response to authenticate call
+    Given I provide the following valid data:
+      | property  | value       |
+      | username  | shep_clark  |
+      | password  | govols!!!   |
+    When I request "/authentication" be created
+    Then the following data is returned:
+      | property      | value                 |
+      | employee_id   | 123                   |
+      | method.add    | no                    |
+      | method.review | no                    |
+      | mfa.add       | yes                   |
+      | mfa.review    | no                    |
+    When I request "/authentication" be created
```

Comment:
The presence of multiple `When`'s here is an indicator that this single scenario may be trying to test too many things. A good rule of thumb can be that [a scenario should have only one When step](https://stackoverflow.com/a/19401196).

I don't know whether it's worth the time, but you might consider writing separate, simpler scenarios for checking the whether the user will be nagged for various scenarios. Example:
```gherkin
  Scenario: Nagging a user to set up MFA
    Given I have provided valid credentials
      And I have not been nagged to set up MFA
      And I have not been nagged to set up a recovery method
    When I log in
    Then I should only be nagged to set up MFA

  Scenario: Nagging a user to set up a recovery method
    Given I have provided valid credentials
      And I have been nagged to set up MFA recently
      But I have not been nagged to set up a recovery method
    When I log in
    Then I should only be nagged to set up a recovery method

  Scenario: Not re-nagging a user right away
    Given I have provided valid credentials
      And I have been nagged to set up MFA recently
      And I have been nagged to set up a recovery method recently
    When I log in
    Then I should not be nagged to set up MFA
      And I should not be nagged to set up a recovery method
```

Reply:
Good question. Is it worth the time to rewrite this? IMHO, no, but worth considering.